projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -82,15 +116,41 @@ export class CartesianLegend {
     const legendEntry = select<EnterElement, Series<{}>>(element).append('div').classed('legend-entry', true);
 
     this.appendLegendSymbol(legendEntry);
-
     legendEntry
       .append('span')
       .classed('legend-text', true)
-      .text(series => series.name);
+      .text(series => series.name)
+      .on('click', series => this.updateActiveSeries(series));
+
+    this.updateLegendClassesAndStyle();
 
     return legendEntry;
   }
 
+  private updateLegendClassesAndStyle(): void {
+    const legendElementSelection = select(this.legendElement!);
+
+    // Legend entry symbol
+    legendElementSelection
+      .selectAll('.legend-symbol circle')
+      .style('fill', series =>
+        !this.isThisLegendEntryActive(series as Series<{}>) ? Color.Gray3 : (series as Series<{}>).color
```

Comment:
All these casts should be avoidable, the type on line 131 is incorrect. See line 116 for an example.